.ci/install_kata_kernel.sh

```diff
@@ -13,80 +13,72 @@ set -e
+kernel_arch="x86_64"
+get_kernel_url="https://cdn.kernel.org/pub/linux/kernel"
+tmp_dir=$(mktemp -d)
+
+download_repo() {
+	pushd ${tmp_dir}
+	git clone --depth 1 https://github.com/${repo_owner}/${repo_name}
+	popd
 }
 
-trap cleanup EXIT
-
-function usage() {
-	cat << EOT
-Usage: $0 <version>
-Install the containers clear kernel image <version> from "${repo_owner}/${repo_name}".
-
-version: Use 'latest' to pull latest kernel or a version from "${cc_linux_releases_url}"
-EOT
-
-	exit 1
+get_kernel_version() {
+	pushd $tmp_dir/$repo_name > /dev/null
+	kernel_version=$(grep "Linux/x86" kernel/configs/x86_kata_kvm_* | cut -d' ' -f3 | tail -1)
+	popd > /dev/null
+	if [ -z "$kernel_version" ]; then
+		echo "unknown kernel version"; exit 1
+	else
+		echo ${kernel_version}
+	fi
 }
 
-#Get latest version by checking remote tags
-#We dont ask to github api directly because force a user to provide a GITHUB token
-function get_latest_version {
-	pushd "${fake_repo_dir}" >> /dev/null
-	git init -q
-	git remote add origin  https://github.com/"${repo_owner}/${repo_name}".git
-
-	cc_release=$(git ls-remote --tags 2>/dev/null \
-		| grep -oP '\-\d+\.container'  \
-			| grep -oP '\d+' \
-				| sort -n | \
-					tail -1 )
-
-	tag=$(git ls-remote --tags 2>/dev/null \
-		| grep -oP "v\d+\.\d+\.\d+\-${cc_release}.container" \
-			| tail -1)
-
-	popd >> /dev/null
-	echo "${tag}"
+# download the linux kernel, first argument is the kernel version
+download_kernel() {
+	kernel_version=$1
+	pushd $tmp_dir
+	kernel_tar_file="linux-${kernel_version}.tar.xz"
+	kernel_url="${get_kernel_url}/v$(echo $kernel_version | cut -f1 -d.).x/${kernel_tar_file}"
+	curl -LOk ${kernel_url}
+	tar -xf ${kernel_tar_file}
+	popd
 }
 
-function download_kernel() {
-	local version="$1"
-	arch=$(arch)
-	[ -n "${version}" ] || die "version not provided"
-	[ "${version}" == "latest" ] && version=$(get_latest_version)
-	echo "version to install ${version}"
-	local binaries_dir="${version}-binaries"
-	local binaries_tarball="${binaries_dir}.tar.gz"
-	local shasum_file="SHA512SUMS"
-	if [ "$arch" = x86_64 ]; then
-		curl -OL "${linux_releases_url}/download/${version}/${binaries_tarball}"
-		curl -OL "${linux_releases_url}/download/${version}/${shasum_file}"
-		sha512sum -c "${shasum_file}"
-		tar xf "${binaries_tarball}"
-	else
-        	die "Unsupported architecture: $arch"
-	fi
-
-	pushd "${binaries_dir}"
-	sudo make install
+# build the linux kernel, first argument is the kernel version
+build_and_install_kernel() {
+	kernel_version=$1
+	pushd ${tmp_dir}
+	kernel_config_file=$(realpath ${repo_name}/kernel/configs/x86_kata_kvm_* | tail -1)
+	kernel_patches=$(realpath ${repo_name}/kernel/patches/*)
+	kernel_src_dir="linux-${kernel_version}"
+	pushd ${kernel_src_dir}
+	cp ${kernel_config_file} .config
+	for p in ${kernel_patches}; do patch -p1 < $p; done
+	make -s ARCH=${kernel_arch} oldconfig > /dev/null
+	make -s ARCH=${kernel_arch} -j$(nproc) > /dev/null
```

Comment:
@grahamwhaley might want to check this line. I guess it's fine under the CI, but it can tend to DoS a system which might not be ideal if this is run locally ;) Hence, we could say to only specify `-j` if `CI=true` (and `KATA_DEV_MODE` is not set) maybe?